SwanProjects/swanprojects/handlers.py

```diff
@@ -0,0 +1,289 @@
+        config=True,
+        help="The path to the JSON containing stack configuration")
+
+class ProjectInfoHandler(APIHandler):
+    @tornado.web.authenticated
+    def post(self):
+        """
+        Post request for the SwanLauncher/SwanFileBrowser,
+        this endpoint returns project information such as stack, release, platform etc..
+        if the path is not inside the project return and empty project data.
+        """
+        input_data = self.get_json_body()
+        self.log.info(f"ProjectInfoHandler = {input_data}")
+        path = input_data["path"]
+
+        project = get_project_path(path)
+        project_data = {}
+        if project is not None:
+            project_data = get_project_info(project)
+
+            project_data["name"] = project.split(os.path.sep)[-1]
+            readme = get_project_readme(project)
+            if readme is not None:
+                project_data["readme"] = readme
+            project_data["user_script"] = get_user_script_content(project)
+        payload = {"project_data": project_data}
+        self.finish(json.dumps(payload))
+
+class StacksInfoHandler(APIHandler):
+
+    swan_projects_config = None
+
+    def initialize(self):
+        self.swan_projects_config = SwanProjects(config=self.config)
+
+    @tornado.web.authenticated
+    def get(self):
+        """
+        This endpoint is required for the project dialog, it's returning the information saved on stacks.json
+        """
+        with open(self.swan_projects_config.stacks_path) as f:
+            stacks = json.loads(f.read())
+        self.finish(json.dumps({"stacks": stacks}))
+
+class KernelSpecManagerPathHandler(APIHandler):
+    @tornado.web.authenticated
+    def post(self):
+        """
+        This endpoint is required for the project kernel spec manager, it's it is setting the path to
+        check if we are inside a project to change the kernel spec manager path
+        """
+        input_data = self.get_json_body()
+        path = input_data["path"]
+
+        self.log.info(f"KernelSpecManagerPathHandler = {input_data}")
+
+        project = get_project_path(path)
+        if self.kernel_spec_manager.set_path(path):
+            data = {"status": True, "is_project": project is not None,
+                    "msg": f"SWAN kernel spec manager set to path: {path}"}
+            self.finish(json.dumps(data))
+        else:
+            data = {"status": False, "is_project": project is not None,
+                    "msg": f"Error setting SWAN kernel spec manager to path: {path}"}
+            self.finish(json.dumps(data))
+
+class CreateProjectHandler(APIHandler):
+    @tornado.web.authenticated
+    def post(self):
+        """
+        Endpoint to create a project, receive project information such as name, stack, platform, release, user_script.
+        The project is created at $HOME/SWAN_projects/project_name and a hidden json ".swanproject" file with the information
```

Comment:
To be changed when projects can be created anywhere.

Reply:
Yes, agree!